### ISSUE TITLE
Internal Error reported in method “allreduce”， Requested ReadyEvent with GPU device but not compiled with CUDA

### ISSUE THREAD

**Issue by jihuacao** [horovod/horovod#2478]
**Environment:**
Framework: PyTorch
Framework version:1.7
Horovod version:0.19.5
MPI version:4.0.4
CUDA version:11.1
NCCL version:2.8.3
Python version:3.6.2
Spark / PySpark version:None
OS and version:ubuntu 16.4
GCC version:5.4.0
CMake version:3.18.4

install command：
HOROVOD_GPU=CUDA HOROVOD_WITH_PYTORCH=1 HOROVOD_GPU_OPERATIONS=NCCL HOROVOD_WITHOUT_GLOO=1 HOROVOD_WITH_MPI=1 HOROVOD_CUDA_HOME=/usr/local/cuda proxychains pip install --no-cache-dir horovod==0.19.5 1>install_hvd_0.19.5.log 2>&1

install log file：
[install_hvd_0.19.5.log](https://github.com/horovod/horovod/files/5588167/install_hvd_0.19.5.log)

py error stack：
Exception has occurred: RuntimeError
Internal error. Requested ReadyEvent with GPU device but not compiled with CUDA.
  File "/data2/process_data/caojihua/workspace/DModel/model/train.py", line 130, in all_reduce
    avg_tensor = hvd.allreduce(val, name=name)
  File "/data2/process_data/caojihua/workspace/DModel/model/train.py", line 97, in train_evaluate_common
    rloss = all_reduce(_loss, 'loss');rclass_loss = all_reduce(_class_loss, 'class_loss');rwh_loss = all_reduce(_wh_loss, 'wh_loss');roffset_loss = all_reduce(_offset_loss, 'offset_loss');riou_loss = all_reduce(_iou_loss, 'iou_loss');mobj_acc = all_reduce(_obj_acc, 'obj_acc');motp_acc = all_reduce(_otp_acc, 'otp_acc');mglobal_acc = all_reduce(_global_acc, 'global_acc')
  File "/data2/process_data/caojihua/workspace/DModel/model/train.py", line 124, in train
    ret = train_evaluate_common(Stage.Train, train_loader, train_sampler, epoch)
  File "/data2/process_data/caojihua/workspace/DModel/model/train.py", line 486, in <module>
    train_ret = train(epoch)

py env：
[envs.log](https://github.com/horovod/horovod/files/5588174/envs.log)

**Comment by tgaddair** [comment]
Hey @jihuacao, looks like GPU support was not built into Horovod.  My guess is that you installed Horovod without GPU support, then attempted to reinstall without ignoring the cache.

Can you try the following:

```
pip uninstall horovod
[ENV VARIABLES] pip install --no-cache-dir horovod==0.19.5
horovodrun --check-build
```

**Comment by jihuacao** [comment]
Thanks for your reply @tgaddair 
before specify nccl env:
![before_specify_nccl_env](https://user-images.githubusercontent.com/22140223/100191171-aa772900-2f2a-11eb-93cd-42b5c89e4c74.png)
The Internal error exists.
after specify nccl:
![after_specify_nccl_env](https://user-images.githubusercontent.com/22140223/100191238-bf53bc80-2f2a-11eb-8559-e80b83c5f9fb.png)
The Internal error exists.

Is there any wrong in installation procedure?

I read a little bit of the setup.py and the horovod/torch/ready_event.cc. It seem that macro HAVE_GPU is not set which cause this internal error, and the HAVE_GPU is depened on the function test_compile() in setup.py. I think: Is horovod-0.19.5 compatible with torch-1.7?

Compile the torch:origin/master with cuda11.1-->reinstall the horovod:
torch block in model.cuda(),  pause and maybe bad end.

**Comment by jihuacao** [comment]
@tgaddair 
![image](https://user-images.githubusercontent.com/22140223/100200714-f4b3d680-2f39-11eb-9803-ce35b2c10f89.png)
![image](https://user-images.githubusercontent.com/22140223/100200765-0b5a2d80-2f3a-11eb-9254-fdda584fbeff.png)

**Comment by jihuacao** [comment]
Very sorry! the same mistake with #2474 I made.
